clock_control: stm32: Configure MCO through devicetree instead of Kconfig (#76556)
zephyrproject-rtos/zephyr: doc/releases/release-notes-4.0.rst

  * :ref:`native_posix<native_posix>` has been deprecated in favour of
    :ref:`native_sim<native_sim>`.
  * Support for Google Kukui EC board (``google_kukui``) has been dropped.

>   * STM32: Deprecated setting MCO source and prescalor in Kconfig in favour of setting it through devicetree.
>     See ``samples/boards/stm32/mco`` sample.

Reviewer: Nit:
```suggestion
  * STM32: Deprecated MCO configuration via Kconfig in favour of setting it through devicetree.
    See ``samples/boards/stm32/mco`` sample.
```
